pkg/trace/agent/agent.go

```diff
@@ -651,23 +653,35 @@ func (a *Agent) runSamplers(now time.Time, ts *info.TagStats, pt traceutil.Proce
 	rare := a.RareSampler.Sample(now, pt.TraceChunk, pt.TracerEnv)
 
 	if a.conf.ProbabilisticSamplerEnabled {
+		samplerName = sampler.NameProbabilistic
 		if rare {
+			samplerName = sampler.NameRare
 			return true, true
 		}
 		if a.ProbabilisticSampler.Sample(pt.Root) {
 			pt.TraceChunk.Tags[tagDecisionMaker] = probabilitySampling
 			return true, true
 		}
 		if traceContainsError(pt.TraceChunk.Spans, false) {
+			samplerName = sampler.NameError
 			return a.ErrorsSampler.Sample(now, pt.TraceChunk.Spans, pt.Root, pt.TracerEnv), true
 		}
 		return false, true
 	}
 
 	priority, hasPriority := sampler.GetSamplingPriority(pt.TraceChunk)
 	if hasPriority {
+		samplerName = sampler.NamePriority
+		samplingPriority = priority
+		if spans := pt.TraceChunk.GetSpans(); len(spans) > 0 && spans[0].Meta[tagDecisionMaker] == probabilitySampling {
+			// This trace has already been sampled by the probabilistic sampler in the OTLPReceiver.
+			// When sampler.PriorityAutoKeep is set as the trace chunk priority,
+			// the probabilistic sampler assigns the decision maker (probabilitySampling).
+			samplerName = sampler.NameProbabilistic
+		}
```

Comment:
To @ajgajg1134 's  https://github.com/DataDog/datadog-agent/pull/33695#discussion_r1949670116, 

This logic is based on 

https://github.com/DataDog/datadog-agent/blob/e007f2d8e032ef430a3979e27af2811c28d3d685/pkg/trace/api/otlp.go#L494-L536

Reply:
hmm, I'm not confident we need this code. Since if that decision maker was set it would mean the probabilistic sampler was enabled which means this function would have returned at line 663? in other words, otlp will only set that DM if the probabilistic sampler is enabled meaning we shouldn't ever reach this line?

If I'm wrong about the above then one small different note: Sorry I didn't notice before, but I believe it would be preferred to check the trace chunk for this tag and not just look at the very first span (as we're _very slowly_ trying to move folks to using tags at the right "level").